frontend/src/component/admin/roles/RoleForm/RolePermissionCategories/RolePermissionProject.tsx

```diff
@@ -0,0 +1,148 @@
+    const allPermissions = useMemo(
+        () =>
+            permissions?.map((permission) => getRoleKey(permission)).sort() ||
+            [],
+        [permissions],
+    );
+
+    const otherPermissions = useMemo(() => {
+        const allStructuredPermissions = PROJECT_PERMISSIONS_STRUCTURE.flatMap(
+            (category) =>
+                category.permissions.map(([permission]) => permission),
+        ).sort() as string[];
+
+        return allPermissions.filter(
+            (permission) => !allStructuredPermissions.includes(permission),
+        );
+    }, [permissions]);
+
+    const permissionsStructure = useMemo(
+        () =>
+            [
+                ...PROJECT_PERMISSIONS_STRUCTURE,
+                {
+                    label: 'Other',
+                    permissions: otherPermissions.map((p) => [p]),
+                } as PermissionCategory,
+            ]
+                .map((category) => ({
+                    label: category.label,
+                    permissions: category.permissions
+                        .filter(([permission]) =>
+                            allPermissions.includes(permission),
+                        )
+                        .map(([permission, parentPermission]) => [
+                            permissions.find(
+                                (p) => getRoleKey(p) === permission,
+                            ),
+                            parentPermission,
+                        ]) as [IPermission, string?][],
+                }))
+                .filter((category) => category.permissions.length > 0),
+        [allPermissions],
+    );
+
+    return (
+        <StyledGrid>
+            {permissionsStructure.map((category) => (
+                <div>
+                    <StyledSectionTitle>{category.label}</StyledSectionTitle>
+                    <div>
+                        {category.permissions.map(
+                            ([permission, parentPermission]) => (
+                                <Box
+                                    sx={(theme) => ({
+                                        marginLeft: parentPermission
+                                            ? theme.spacing(1.5)
+                                            : 0,
+                                    })}
+                                >
+                                    <FormControlLabel
+                                        data-testid={permission}
+                                        key={permission.name}
+                                        control={
+                                            <Checkbox
+                                                checked={Boolean(
+                                                    checkedPermissions[
+                                                        permission.name
+                                                    ] ||
+                                                        (parentPermission &&
```

Comment:
with this level of nesting it becomes hard to read. I'd extract a component

Reply:
refactored